### DISCUSSION TITLE
bug with prisma findmany() on vercel

### DISCUSSION THREAD

**Discussion by nitlott** [vercel/community#2468] [Help] [answered]
### Summary

this snippet `import { PrismaClient } from "@prisma/client";
const prisma = new PrismaClient();
export const GET = async (req: Request, res: Response) => {
  try {
    const prompts = await prisma.prompt.findMany({
      include: { creator: true },
    });
    return new Response(JSON.stringify(prompts), { status: 200 });
  } catch (error) {
    console.log("Error: ", error);
    return new Response(
      JSON.stringify({ message: "Failed to fetch all prompts" }),
      { status: 500 }
    );
  }
`};`

works perfectly in localenv pointing to same database as in production as on vercel, however on vercel it does not work good at all. it only gets data once then it stops working. whats the issue with prisma and vercel?
all other prisma parts but not this. and seams related with vercels enviroment caching the response. 

not even when visiting the endpoint directly it shows newly created posts, only old and also the deleted ones. i tried multiple browsers and devices and it seam to keep the first load on all devices wierdly enoug.... in local env it works perfectly, so not a database issue since same database. seams related to vercel only. same results using NextResponse, NextRequest.

this occured when implementing prisma, i have all setup in config to generate tables etc so thats correct too.
bug?

### Additional information

database is supabase

**Comment by nitlott** [comment, marked as answer]
after more investigation it seams it is prismas findmany in combination with nextjs, it does not mather if its on vercel or not.
if i npm run build and npm run start locally i get same results.
but if i run npm run dev it works.

this findmany caches forever after first run (even if you try other devices)
`const prompts = await prisma.prompt.findMany({
      include: { creator: true },
    });`


and this one works without a problem everytime
    `const prompts = await prisma.prompt.findMany({
      where: {
        creatorId: params.id,
      },
      include: { creator: true },
    });`

**Reply by amyegan** [reply]
Thanks for sharing what you found. I see you also opened a new issue in the Next.js repo, so the team is following up there 🙂 

https://github.com/vercel/next.js/issues/49854

**Comment by nitlott** [comment]
i did a little workaround in this way, might help someoneelse:
`  
const page: number = Math.min(
    Math.max(parseInt(req.nextUrl.searchParams.get("page") || "1", 10), 1),
    1000
  ); // Limit to a maximum of 1000 pages

  const pageSize: number = Math.min(
    parseInt(req.nextUrl.searchParams.get("pageSize") || "10", 10),
    100
  ); // Limit page size to a maximum of 100 items

  const skip = (page - 1) * pageSize;

  const take = pageSize;

  try {
    const prompts = await prisma.prompt.findMany({
      skip,
      take,
      include: { creator: true },
    });
`
